src/pytorch/CFConvNeighbors.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2020-2021 Acellera
+ * Authors: Raimondas Galvelis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <torch/script.h>
+#include "CFConv.h"
+
+#ifndef NNPOPS_CFCONVNEIGHBORS
+#define NNPOPS_CFCONVNEIGHBORS
+
+namespace NNPOps {
+namespace CFConvNeighbors {
+
+using torch::Device;
+using torch::Tensor;
```

Comment:
The `using` keyword should never be used in a header file.  If you do, it pollutes the namespace of any source file that includes it.

Reply:
These are within a namespace, so the mapping is `torch::Device` --> `NNPOps::CFConvNeighbors::Device`, not just `Device`. Also, this header is internal and would be needed for the public API.

Reply:
I still wouldn't do it.  This is just one of those rules of C++ development that it's best to follow consistently.  There's little to be gained from breaking it and a lot to lose.  Once you write a header file, you never know where someone else might include that file in the future or what assumptions they'll have made.

Reply:
Removed that stuff